target/linux/ath79/dts/qca9558_belkin_f9j1108-v2.dts

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca955x.dtsi"
+
+/ {
+	model = "Belkin AC1750 DB Wi-Fi (F9J1108v2)";
+	compatible = "belkin,f9j1108-v2", "qca,qca9558";
+
+	aliases {
+		led-boot = &led_system;
+		led-failsafe = &led_system;
+		led-running = &led_system;
+		led-upgrade = &led_system;
+		label-mac-device = &eth1;
+	};
+	
+	leds {
+		compatible = "gpio-leds";
+
+		led_system: system {
+			label = "belkin:blue:system";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+		};
+
+		usb {
+			label = "belkin:blue:usb";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&hub_port1>;
+			linux,default-trigger = "usbport";
+		};
+
+		wps-blue {
+			label = "belkin:blue:wps";
+			gpios = <&gpio 19 GPIO_ACTIVE_LOW>;
+		};
+		
+		wps-amber {
+			label = "belkin:amber:wps";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+	};
+	
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		gpio_usb_power {
+			gpio-export,name = "belkin:usb-power";
+			gpio-export,output = <1>;
+			gpios = <&gpio 21 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+		
+		wps {
+			label = "wps";
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&pcie1 {
+	status = "okay";
+};
+
+&uart {
+	status = "okay";
+};
+
+&usb_phy0 {
+	status = "okay";
+};
+
+&usb0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port0: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
```

Comment:
No trigger for this one?

Reply:
From what I can tell, hub_port0 isn't triggered by GPIO LED, but only hub_port1 is.

Reply:
in the description you mention two USB 3.0 ports
&usb0 and &usb1 are two EHCI (= USB 2.0) controllers

are you sure those are used? on the European hardware (F9K1115v2) there are two USB ports:
- one USB 3.0 port connected to the EJ168 USB 3.0 Host Controller
- one USB 2.0 port connected to &usb0
- &usb1 is not used

(ar71xx always enabled both SoC internal USB ports - which are &usb0 and &usb1. it will still steal some RAM but not hurt if still enabled)

Reply:
You are indeed correct, &usb1 is not used, debugging on both ports shows the following:

USB 1 (bottom) USB 2.0 (EHCI) connected to the SoC directly.

/sys/devices/platform/ahb/1b000000.usb/usb1/1-0:1.0

[53759.542331] usb 1-1: new high-speed USB device number 2 using ehci-platform
[53759.706102] usb-storage 1-1:1.0: USB Mass Storage device detected
[53759.713361] usb-storage 1-1:1.0: Quirks match for vid 054c pid 02a5: 80000
[53759.720740] scsi host0: usb-storage 1-1:1.0
[53761.332086] scsi 0:0:0:0: Direct-Access     Sony     Storage Media    0100 PQ: 0 ANSI: 0 CCS
[53761.344404] sd 0:0:0:0: [sda] 31303680 512-byte logical blocks: (16.0 GB/14.9 GiB)
[53761.352743] sd 0:0:0:0: [sda] Write Protect is off
[53761.358498] sd 0:0:0:0: [sda] No Caching mode page found
[53761.363920] sd 0:0:0:0: [sda] Assuming drive cache: write through
[53761.388079]  sda: sda1 sda2
[53761.395377] sd 0:0:0:0: [sda] Attached SCSI removable disk

USB 2 (top) - USB 3.0 (XHCI) connected via PCI

/sys/devices/pci0000:01/0000:01:00.0/usb3/3-0:1.0

[53789.857413] usb 1-1: USB disconnect, device number 2
[53791.594330] usb 3-1: new high-speed USB device number 2 using xhci_hcd
[53791.752689] usb-storage 3-1:1.0: USB Mass Storage device detected
[53791.759846] usb-storage 3-1:1.0: Quirks match for vid 054c pid 02a5: 80000
[53791.767208] scsi host0: usb-storage 3-1:1.0
[53793.364254] scsi 0:0:0:0: Direct-Access     Sony     Storage Media    0100 PQ: 0 ANSI: 0 CCS
[53793.376122] sd 0:0:0:0: [sda] 31303680 512-byte logical blocks: (16.0 GB/14.9 GiB)
[53793.384405] sd 0:0:0:0: [sda] Write Protect is off
[53793.390082] sd 0:0:0:0: [sda] No Caching mode page found
[53793.395519] sd 0:0:0:0: [sda] Assuming drive cache: write through
[53793.416675]  sda: sda1 sda2
[53793.423853] sd 0:0:0:0: [sda] Attached SCSI removable disk